src/openforms/submissions/tasks/cleanup.py

```diff
@@ -46,12 +46,18 @@ def cleanup_on_completion_results():
 
     cutoff = timezone.now() - timedelta(days=RETAIN_RESULTS_NUM_DAYS)
 
-    submissions = Submission.objects.filter(
-        # only consider submissions that are completed and old enough
-        completed_on__lte=cutoff,
-        suspended_on__isnull=True,
-        # only clean up submissions that have task_ids stored (prevent re-processing)
-        on_completion_task_ids__len__gt=0,
+    submissions = (
+        Submission.objects.annotate(
+            number_tasks_ids=F("postcompletiontasksids__tasks_ids__len")
```

Comment:
how does this `__len` work if there are multiple records for the same submission? Can `__len` result in `None` for some records?

Reply:
There is this test in src/openforms/submissions/tests/test_submission_status.py which tests the scenario of multiple records:
```python
def test_cleanup_for_submission_with_multiple_post_completion_events(
        self, mock_forget
    ):
        submission = SubmissionFactory.create(
            completed=True,
            completed_on=timezone.now() - timedelta(days=2, seconds=10),
            suspended_on=None,
        )
        PostCompletionMetadataFactory.create(
            submission=submission, tasks_ids=["some-id-1", "some-id-2"]
        )
        PostCompletionMetadataFactory.create(
            submission=submission, tasks_ids=["some-id-3"]
        )
        PostCompletionMetadataFactory.create(submission=submission, tasks_ids=[])

        cleanup_on_completion_results()

        self.assertEqual(3, mock_forget.call_count)
```
I will add a test for the case without record!